src/vaults/factories/PoolEscrowFactory.sol

```diff
@@ -76,7 +80,8 @@ contract PoolEscrowFactory is IPoolEscrowFactory, Auth {
     // --- Internal methods ---
     function _deterministicAddress(uint64 poolId) internal view returns (address) {
         bytes32 salt = bytes32(uint256(poolId));
-        bytes memory bytecode = abi.encodePacked(type(PoolEscrow).creationCode, abi.encode(poolId, address(this)));
+        bytes memory bytecode =
+            abi.encodePacked(type(PoolEscrow).creationCode, abi.encode(poolId, sharedGateway, address(this)));
```

Comment:
IIUC, shared dependency cannot be assumed to be static right? If so, this would be a breaking change for this deterministic address retrieval.

Reply:
Yes! it's guaranteed to be static. That's why I added the SharedDependency. If Gateway changes, we just change the gateway address inside `SharedDependency`, without touching the rest of the consumers, aka, pool escrows.